Include/lock.h

```diff
@@ -0,0 +1,16 @@
+#ifndef Py_LOCK_H
+#define Py_LOCK_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef Py_LIMITED_API
+#  define Py_CPYTHON_LOCK_H
+#  include "cpython/lock.h"
```

Comment:
I don't see the value of such header file. Just include cpython/lock.h in Python.h, and check Py_LIMITED_API in cpython/lock.h.

Reply:
@ericsnowcurrently expressed a preference for this style when `pyatomic.h` was added: https://github.com/python/cpython/pull/109344#discussion_r1325076002

Reply:
IMO, it's helpful to follow a consistent pattern when it comes to the Include/cpython header files.  That means in some cases we end up with very minimal header files like this in Include/.

Reply:
If you mention consistency, there are already many header files in Include/cpython/ which have no companion Include/ header file:

```
$ grep cpython/ Include/Python.h
#include "cpython/pydebug.h"
#include "cpython/longintrepr.h"
#include "cpython/odictobject.h"
#include "cpython/funcobject.h"
#include "cpython/classobject.h"
#include "cpython/code.h"
#include "cpython/cellobject.h"
#include "cpython/initconfig.h"
#include "cpython/genobject.h"
#include "cpython/picklebufobject.h"
#include "cpython/pytime.h"
#include "cpython/context.h"
#include "cpython/pyctype.h"
#include "cpython/pyfpe.h"
#include "cpython/tracemalloc.h"
#include "cpython/optimizer.h"
```